spec/commands/update_spec.rb

```diff
@@ -421,6 +421,39 @@
     expect(the_bundle).to include_gems "thin 2.0", "rack 10.0", "rack-obama 1.0"
   end
 
+  it "will not warn when an explicitly updated git gem changes sha but not version" do
```

Comment:
this is the test that's failing, and I'm not 100% how to refactor to fix it

Reply:
Fixed the test by saving the string representation of each source _before_ it's actually updated, so the comparison later works as expected.